src/main/java/core/basesyntax/stategy/handler/PurchaseOperation.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.stategy.handler;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void process(FruitTransaction transaction) {
+        Integer fruitQuantity = Storage.fruits.get(transaction.getFruitName());
```

Comment:
There is a possibility of encountering a null value when retrieving 'fruitQuantity' from the Storage. You should check if 'fruitQuantity' is non-null before proceeding with operations on it.